handler/handler_test.go

```diff
@@ -0,0 +1,437 @@
+				},
+			}
+			instance = EnqueueRequestForAnnotation{Type: schema.GroupKind{Group: "apps", Kind: "ReplicaSet"}}
+
+			evt := event.CreateEvent{
+				Object: repl,
+				Meta:   repl.GetObjectMeta(),
+			}
+
+			instance.Create(evt, q)
+			Expect(q.Len()).To(Equal(0))
+
+			newRepl := repl.DeepCopy()
+			newRepl.Name = pod.Name + "2"
+			newRepl.Namespace = pod.Namespace + "2"
+
+			newRepl.Annotations = map[string]string{
+				TypeAnnotation:           schema.GroupKind{Group: "apps", Kind: "ReplicaSet"}.String(),
+				NamespacedNameAnnotation: "foo/faz",
+			}
+
+			instance2 := EnqueueRequestForAnnotation{Type: schema.GroupKind{Group: "apps", Kind: "ReplicaSet"}}
+
+			evt2 := event.UpdateEvent{
+				ObjectOld: repl,
+				MetaOld:   repl.GetObjectMeta(),
+				ObjectNew: newRepl,
+				MetaNew:   newRepl.GetObjectMeta(),
+			}
+
+			instance2.Update(evt2, q)
+			Expect(q.Len()).To(Equal(1))
+
+			i, _ := q.Get()
+			Expect(i).To(Equal(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: "foo",
+					Name:      "faz",
+				},
+			}))
+		})
+		It("should enqueue a Request to the owner resource when the annotations are applied in child object"+
+			"in the Create Event", func() {
+			repl := &appsv1.ReplicaSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "foo",
+					Name:      "faz",
+				},
+			}
+
+			err := SetOwnerAnnotations(podOwner, repl)
+			Expect(err).To(BeNil())
+
+			evt := event.CreateEvent{
+				Object: repl,
+				Meta:   repl.GetObjectMeta(),
+			}
+
+			instance.Create(evt, q)
+			Expect(q.Len()).To(Equal(1))
+
+			i, _ := q.Get()
+			Expect(i).To(Equal(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: podOwner.Namespace,
+					Name:      podOwner.Name,
+				},
+			}))
+		})
+
+		It("should enqueue a Request with the annotations of the object in case of GenericEvent", func() {
+			evt := event.GenericEvent{
+				Object: pod,
+				Meta:   pod.GetObjectMeta(),
+			}
+			instance.Generic(evt, q)
+			Expect(q.Len()).To(Equal(1))
+
+			i, _ := q.Get()
+			Expect(i).To(Equal(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: podOwner.Namespace,
+					Name:      podOwner.Name,
+				},
+			}))
+		})
+
+		It("should not enqueue a request if there are no annotations matching with the object", func() {
+			repl := &appsv1.ReplicaSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "foo",
+					Name:      "faz",
+				},
+			}
+
+			evt := event.CreateEvent{
+				Object: repl,
+				Meta:   repl.GetObjectMeta(),
+			}
+
+			instance.Create(evt, q)
+			Expect(q.Len()).To(Equal(0))
+		})
+
+		It("should not enqueue a Request if there is no Namespace and name annotation matching the specified object are found", func() {
+			repl := &appsv1.ReplicaSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "foo",
+					Name:      "faz",
+					Annotations: map[string]string{
+						TypeAnnotation: schema.GroupKind{Group: "core", Kind: "Pods"}.String(),
```

Comment:
Should this be the following?
```suggestion
						TypeAnnotation: schema.GroupKind{Group: "", Kind: "Pod"}.String(),
```

See https://github.com/kubernetes/api/blob/e4973e079a1129e83001d8368911eec19d9231fa/core/v1/register.go#L26